kafka-ui-react-app/src/components/Schemas/Edit/Edit.tsx

```diff
@@ -70,11 +72,16 @@ const Edit = ({
         clusterName,
         subject
       );
-      history.push(clusterSchemaPath(clusterName, subject));
     },
     [schema, register, control, clusterName, subject, updateSchema, history]
   );
 
+  React.useEffect(() => {
+    if (isSubmitting && isSchemaUpdated) {
+      history.push(clusterSchemaPath(clusterName, subject));
+    }
+  }, [isSchemaUpdated, clusterName, subject]);
```

Comment:
This looks confusing. `isSubmitting` is missing in the dependencies of this effect. If someone will add it in the future, functionality might break, since it's expected that `isSubmitting` to be false when schema is updated.